homeassistant/components/gree/switch.py

```diff
@@ -16,25 +20,52 @@
 from .const import COORDINATORS, DISPATCH_DEVICE_DISCOVERED, DISPATCHERS, DOMAIN
 from .entity import GreeEntity
 
-GREE_SWITCHES: tuple[SwitchEntityDescription, ...] = (
-    SwitchEntityDescription(
+
+@dataclass
+class GreeRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    get_value_fn: Callable[[Device], bool]
+    set_value_fn: Callable[[Device, bool], None]
+
+
+@dataclass
+class GreeSwitchEntityDescription(SwitchEntityDescription, GreeRequiredKeysMixin):
+    """Describes Gree switch entity."""
+
+
+GREE_SWITCHES: tuple[GreeSwitchEntityDescription, ...] = (
+    GreeSwitchEntityDescription(
         icon="mdi:lightbulb",
         name="Panel Light",
         key="light",
+        get_value_fn=lambda d: d.light,
+        set_value_fn=lambda d, v: setattr(d, "light", v),
```

Comment:
`setattr` on a static string isn't safer compared to just calling it.

```suggestion
        set_value_fn=lambda d, v: d.light = v,
```

Reply:
Your suggestion doesn't work - it generates a SyntaxError.
Right now the only other possibility I see is to create 5 separate "referenced" functions.

Reply:
Oh right, doh 🤦